scst-store/create-service-account-access-token.hbs.md

```diff
@@ -16,108 +16,243 @@ You can create two types of service accounts:
 
 ### <a id='ro-serv-accts'></a>Read-only service account
 
-As a part of the Store installation, the `metadata-store-read-only` cluster role is created by default. This cluster role allows the bound user to have `get` access to all resources. To bind to this cluster role, run the following command:
+As a part of the Store installation, the `metadata-store-read-only` cluster role is created by default. This cluster role allows the bound user to have `get` access to all resources. To bind to this cluster role, run the following command depending on the k8s version:
 
-```console
-kubectl apply -f - -o yaml << EOF
----
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
-metadata:
-  name: metadata-store-ready-only
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: metadata-store-read-only
-subjects:
-- kind: ServiceAccount
-  name: metadata-store-read-client
-  namespace: metadata-store
----
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: metadata-store-read-client
-  namespace: metadata-store
-automountServiceAccountToken: false
-EOF
-```
+* Kubernetes version before 1.24
+    ```console
+    kubectl apply -f - -o yaml << EOF
+    ---
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRoleBinding
+    metadata:
+      name: metadata-store-ready-only
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: ClusterRole
+      name: metadata-store-read-only
+    subjects:
+    - kind: ServiceAccount
+      name: metadata-store-read-client
+      namespace: metadata-store
+    ---
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: metadata-store-read-client
+      namespace: metadata-store
+    automountServiceAccountToken: false
+    EOF
+    ```
+* Kubernetes version 1.24 or later
+    ```console
+    kubectl apply -f - -o yaml << EOF
+    ---
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRoleBinding
+    metadata:
+      name: metadata-store-ready-only
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: ClusterRole
+      name: metadata-store-read-only
+    subjects:
+    - kind: ServiceAccount
+      name: metadata-store-read-client
+      namespace: metadata-store
+    ---
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: metadata-store-read-client
+      namespace: metadata-store
+    annotations:
+        kapp.k14s.io/change-group: "metadata-store.apps.tanzu.vmware.com/service-account"
+    automountServiceAccountToken: false
+    ---
+    apiVersion: v1
+    kind: Secret
+    type: kubernetes.io/service-account-token
+    metadata:
+        name: metadata-store-read-client
+        namespace: metadata-store
+    annotations:
+        kapp.k14s.io/change-rule: "upsert after upserting metadata-store.apps.tanzu.vmware.com/service-account"
```

Comment:
I think the kapp annotations won't be necessary since this kubectl apply command will be applied after the metadata-store is already installed.

Reply:
oh I thought this annotation was there to avoid a race condition where the secret is being created before the service account, so then it will complain that the service account doesn't exist

Reply:
Yeah, you're right. I take it back!